packages/emotion-theming/src/theme-provider.js

```diff
@@ -42,19 +45,15 @@ type Props = {
 }
 
 let ThemeProvider = (props: Props) => {
+  let theme = useContext(ThemeContext)
+
+  if (props.theme !== theme) {
+    theme = createCacheWithTheme(theme)(props.theme)
```

Comment:
seems like this could be memoized to improve perf for nested ThemeProviders etc

Reply:
It's already memoized with weak maps

Reply:
Oh, you are right, cool!